azext_edge/edge/providers/check/mq.py

```diff
@@ -539,14 +625,349 @@ def evaluate_brokers(
+
+    check_manager = CheckManager(
+        check_name="evalBrokerAuthentications",
+        check_desc="Evaluate MQTT Broker Authentications",
+    )
+
+    target_authentications = "brokerauthentications.mqttbroker.iotoperations.azure.com"
+    auth_conditions = ["len(spec.authenticationMethods)"]
+    all_authentications = get_resources_by_name(
+        api_info=MQ_ACTIVE_API,
+        kind=MqResourceKinds.BROKER_AUTHENTICATION,
+        resource_name=resource_name,
+    )
+
+    if not all_authentications:
+        status = CheckTaskStatus.skipped.value if resource_name else CheckTaskStatus.error.value
+        fetch_authentications_error_text = (
+            f"Unable to fetch {MqResourceKinds.BROKER_AUTHENTICATION.value}s in any namespace."
+        )
+        check_manager.add_target(target_name=target_authentications)
+        check_manager.add_target_eval(
+            target_name=target_authentications,
+            status=status,
+            value=fetch_authentications_error_text,
+        )
+        check_manager.add_display(
+            target_name=target_authentications,
+            display=Padding(fetch_authentications_error_text, (0, 0, 0, DEFAULT_PADDING)),
+        )
+        return check_manager.as_dict(as_list)
+
+    for namespace, authentications in get_resources_grouped_by_namespace(all_authentications):
+        check_manager.add_target(
+            target_name=target_authentications,
+            namespace=namespace,
+            conditions=auth_conditions,
+        )
+        check_manager.add_display(
+            target_name=target_authentications,
+            namespace=namespace,
+            display=Padding(
+                f"Broker Authentications in namespace {{{colorize_string(color='purple', value=namespace)}}}",
+                (0, 0, 0, DEFAULT_PADDING),
+            ),
+        )
+
+        authentications = list(authentications)
+        added_broker_ref_condition = False
+
+        for auth in authentications:
+            auth_metadata = auth["metadata"]
+            auth_name = auth_metadata["name"]
+            # store results for check that will used to display later
+            sub_check_results: List[CheckResult] = []
+
+            # check broker reference
+            broker_ref = auth_metadata.get("ownerReferences", [])
+            ref_display = _evaluate_broker_reference(
+                check_manager=check_manager,
+                owner_reference=broker_ref,
+                target_name=target_authentications,
+                namespace=namespace,
+                resource_name=auth_name,
+                added_condition=added_broker_ref_condition,
+            )
+
+            auth_desc = f"\n- Broker Authentication {{{colorize_string(auth_name)}}}. {ref_display}"
+            check_manager.add_display(
+                target_name=target_authentications,
+                namespace=namespace,
+                display=Padding(auth_desc, (0, 0, 0, DEFAULT_PADDING)),
+            )
```

Comment:
Similar question here to the previous one about `evaluate_broker_ref`: https://github.com/Azure/azure-iot-ops-cli-extension/pull/382/files#r1868441670